File: examples/demo/04/Library.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

public class Library<T> {

  T t;

  private native @Nullable T mayReturnNull();

  public Library() {
    t = mayReturnNull();
  }

  public T get() {
    return t;
  }
}

File: infer/tests/build_systems/incremental_analysis_change_tenv/src/library.c
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

struct my_pair {
  int x;
  int y;
};

struct my_pair make_pair(int x, int y) {
  struct my_pair pair;
  pair.x = x;
  pair.y = y;
  return pair;
}
